src/main/java/org/motechproject/mots/web/UserController.java

```diff
@@ -86,6 +89,35 @@ public UserDto saveUser(@PathVariable("id") UUID id, @RequestBody UserDto userDt
     return userMapper.toDto(userService.saveUser(existingUser));
   }
 
+  /**
+   * Update User password.
+   * @param currentPassword user previous password, should be the same as current password
+   * @param newPassword user's new password
+   * @return updated User
+   */
+  @RequestMapping(value = "/user/password/change/{currentPassword}/{newPassword}",
+      method = RequestMethod.POST)
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public ResponseEntity<String> changeCurrentUserPassword(
+      @PathVariable("currentPassword") String currentPassword,
+      @PathVariable("newPassword") String newPassword) {
+
+    String userName =
+        (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
@jkondrat do you think this is the best way to get the logged in user?

Reply:
It's fine to leave it as is, but we could also add principal to the parameters like so:
> changeCurrentUserPassword(Principal principal) {
>   principal.getName()
> }